src/components/services-view/service-view.classes.ts

```diff
@@ -0,0 +1,193 @@
+                let cat = getCategoryFromSubCategory(subCat);
+                catsBySubCats[subCat] = cat;
+            });
+        }
+        return catsBySubCats;
+    }
+}
+
+export class ImageItem implements IServiceItem {
+    public iconClass: string;
+    public imageUrl: string;
+    public name: string;
+    public catsBySubCats: any;
+    public description: string;
+    public longDescription: string;
+    public data: any;
+    public isBuilder: boolean;
+    private annotations: any;
+    private $filter: any;
+
+    constructor (image: any, $filter: any) {
+        this.data = image;
+        this.annotations = this.getAnnotations();
+        this.iconClass = this.getIcon();
+        this.name = this.getName();
+        this.description = this.getDescription();
+        this.longDescription = this.getLongDescription();
+        this.catsBySubCats = this.getCategoriesBySubCategories();
+        this.$filter = $filter;
+        console.log("construct $filter = " + JSON.stringify($filter, null, ' '));
+    }
+
+    private getAnnotations() {
+        if (this.data.spec && this.data.spec.tags && this.data.spec.tags.length > 1 &&
+            this.data.spec.tags[0].annotations) {
+            return this.data.spec.tags[0].annotations;
+        }
+    }
+
+    private getIcon() {
+        console.log("$filter = " + JSON.stringify(this.$filter, null, ' '));
+        let icon = this.$filter('imageStreamTagIconClass')(this.data, "icon");
+        console.log("icon = " + icon);
+        if (this.annotations) {
+            return 'font-icon ' + this.annotations.iconClass;
+        }
+    }
+
+    private getName() {
+        if (this.data.metadata.annotations) {
+            return this.data.metadata.annotations['openshift.io/display-name'];
+        }
+    }
+
+    // TODO OpenShift Imagestreams desciptions/long descriptions vary depending
+    // on the 'version' (spec.tags[].annotations). For now, hardcoding to first tag/annotation
+
+    private getDescription() {
+        // do not see a short description to use?
+        return null;
+    }
+
+    private getLongDescription() {
+        if (this.annotations) {
+            return this.annotations.description;
+        }
+    }
+
+    private getCategoriesBySubCategories() {
+        let catsBySubCats = {};
+        if (this.annotations) {
+            let tags = this.annotations.tags.split(/\s*,\s*/);
+            this.isBuilder = _.some(tags, (tag) => {
+                return tag === 'builder';
+            });
+            let lastTag = tags.pop();
+            let subCat = getSubCategoryFromAlias(lastTag);
+            let cat = getCategoryFromSubCategory(subCat);
+            catsBySubCats[subCat] = cat;
+            return catsBySubCats;
+        }
+    }
+}
+
+var categories = allCategories;
+
+function getCategoryFromSubCategory(subCategory: string) {
+    let retCategory: string = 'other';
+
+    let breakLoops: boolean = false;
+    _.each(categories, (category) => {
```

Comment:
Fixed this in ''src/services/catalog.service.ts', although I got a runtime error when I tried to use _.findWhere, but it should exist, we are using lodash 3.10.1?  I used _.findIndex instead.